.circleci/config.yml

```diff
@@ -0,0 +1,76 @@
+---
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: debian:jessie
+
+    environment:
+      STACK_VERSION: 1.6.5
+      GHC_8_2_RESOLVER: nightly-2017-12-16
+
+    steps:
+      - checkout:
+          path: ~/stratosphere
+
+      - run:
+          name: Setup Environment Variables
+          command: |
+            echo 'export PATH="$HOME/.local/bin:$PATH"' >> $BASH_ENV
+
+      - run:
+          name: Install System Dependencies
+          command: |
+            apt-get update --assume-yes
+            apt-get install --assume-yes curl build-essential libgmp3-dev
+
+      - run:
+          name: Install Dependencies
+          # yamllint disable-line rule:line-length
+          command: |
+            mkdir -p ~/.local/bin
+            curl -L https://github.com/commercialhaskell/stack/releases/download/v$STACK_VERSION/stack-$STACK_VERSION-linux-x86_64.tar.gz \
+              | tar -xzv -C ~/.local/bin \
+              && ln -vsf ~/.local/bin/stack-*/stack ~/.local/bin/stack
+
+      - run:
+          name: Build
+          command: |
+            stack setup --resolver $GHC_8_2_RESOLVER
+            stack build --resolver $GHC_8_2_RESOLVER stratosphere --test \
+              --only-dependencies --no-run-tests --jobs=1 \
+              --flag stratosphere:-library-only
+            stack setup
+            stack build stratosphere --test --only-dependencies \
+              --no-run-tests --jobs=1 --flag stratosphere:-library-only
+
+      - run:
+          name: Test
+          command: |
+            stack build --resolver $GHC_8_2_RESOLVER stratosphere --pedantic \
+              --test --jobs=1 --flag stratosphere:-library-only
+            stack haddock --resolver $GHC_8_2_RESOLVER stratosphere \
+              --no-haddock-deps --jobs=1 --flag stratosphere:-library-only
+            stack build stratosphere --pedantic --test --jobs=1 \
+              --flag stratosphere:-library-only
+
+      - run:
+          name: Build Docs
+          command: |
+            stack haddock --resolver $GHC_8_2_RESOLVER stratosphere \
+              --no-haddock-deps --jobs=1 --flag stratosphere:-library-only
+            stack haddock stratosphere --no-haddock-deps --jobs=1 \
+              --flag stratosphere:-library-only
+          no_output_timeout: 30m
+
+      - save_cache:
+          key: dependency-cache-v1-{{ .Branch }}-{{ checksum stack.yaml }}
+          paths:
+            - ~/.stack
+
+      - restore_cache:
+          keys:
+            - dependency-cache-v1-{{ .Branch }}-{{ checksum stack.yaml }}
+            - dependency-cache-v1-{{ .Branch }}-
+            - dependency-cache-v1-
```

Comment:
I would leave this default off, so we must _always_ have a matching `stack.yaml`. If that doesn't match, then we want a fresh cache, or else `~/.stack` will start growing a ton.